### DISCUSSION TITLE
Question about Form builder and Form views

### DISCUSSION THREAD

**Discussion by undersound** [area17/twill#2186] [General]
First of all thanks for your wonderful work! 
It really is a delight working with it from both a dev and end user perspective. 

My questions:

1. On Version 3.0.0-rc4 I started making modules without the _--bladeForm_ option(https://twill.io/docs/3.x/form-fields/)  but since I need to use conditional fields and the sidebar option I want to use the the old form view method. I removed the _getForm_ function from _web/app/Http/Controllers/Twill/PageController.php_ and created a form.blade.php in _web/resources/views/admin/pages/form.blade.php_ but I don't see my body field showing up at _admin/pages/1/edit_ Is there more to it?

2. I am bit confused about which syntaxt to use for i.e. an input field form.blade.php. 

Is it
`@formField('input', [
    'name' => 'subtitle',
    'label' => 'Subtitle',
    'maxlength' => 100,
    'required' => true,
    'note' => 'Hint message goes here',
    'placeholder' => 'Placeholder goes here',
])`

or 

`<x-twill::input 
    name="subtitle"
    label="Subtitle"
    :maxlength="100"
    :required="true"
    note="Hint message goes here"
    placeholder="Placeholder goes here" 
/>`

or are both possible?

Thanks in advance for any guidance

**Comment by iedex** [comment]
Hi,

1. Try moving web/resources/views/admin/pages/form.blade.php to web/resources/views/twill/pages/form.blade.php
2. Both should work

**Reply by undersound** [reply]
1. 😬 left over copy paste operation from 2.x project
2. ok

Thanks!

**Reply by ifox** [reply]
Hi @undersound you can actually use the sidebar with controller based forms, and you can use the `BladePartial` "field" for the part of your form where you need conditional fields. Up to you if you want to mix both approaches or not though!

**Reply by undersound** [reply]
Thanks @ifox !

Is it helpful to make a PR to update the 3.x docs then or are there still form views methods not supported?

_This method will still continue to receive improvements as it does not yet support all the features that the form views method has such as fieldset, sidebars and conditional fields._

[https://twill.io/docs/3.x/form-fields/#using-form-views](https://twill.io/docs/3.x/form-fields/#using-form-views)

**Reply by ifox** [reply]
That's on my to-do list to update before the stable release! Thanks for flagging. Happy to review a PR of you have time.

**Reply by ifox** [reply]
Fieldsets and sidebar fields are supported in the form builder, only conditional fields are not at this point.
